schema_editor/app/scripts/schemas/schemas-service.js

```diff
@@ -120,6 +136,59 @@
         }
 
         /**
+         * Inverse of the above; converts a sub-schema into a JSON-Schema that will allow
+         * JSON-Editor to generate the correct form elements for editing the fields in the schema.
+         * @param {object} definition The sub-schema to convert into fields
+         * @return {array} Form schema to allow editing the fields in definition
+         */
+        function schemaFormDataFromDefinition(definition) {
+            var formData = [];
+            _.each(definition.properties, function(schemaField, title) {
+                var fieldData = {
+                    fieldTitle: title
+                };
+                // Iterate over schema keys and take the appropriate action
+                _.each(schemaField, function(value, key) {
+                    switch(key) {
+                        case 'enum':
+                            fieldData.fieldOptions = value;
+                            break;
+                        case 'type': // This is the JSON-Schema 'type', which is not used here.
+                            break;
+                        case 'media': // Also not used
+                            break;
+                        default:
+                            fieldData[key] = value;
+                    }
+                });
+
+                // Handle required fields, which are stored outside the field info
+                if (_.findIndex(definition.required, function(item) {
```

Comment:
This is kinda difficult to read, I'm generally against relatively complicated ifs, although I'm not sure we have any notes about it in any of our js style guides.

Maybe something more along the lines of this:
```
var requiredIndex = _.findIndex(definition.required, function(item) { return item === title; });
fieldData.isRequired = requiredIndex >= 0;
```